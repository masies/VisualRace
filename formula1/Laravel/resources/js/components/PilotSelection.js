import React from 'react';
import FormGroup from '@material-ui/core/FormGroup';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';

class PilotSelection extends React.Component {
	constructor(props) {
    	super(props)
    }

	render() {
	return (
		<FormGroup row>
			<Grid container spacing={1}  direction="column" >
	  	  { 
            Object.entries(this.props.drivers).map( (driver) =>
            	<Grid container key={driver[0]} direction="column" align="center" style={{marginBottom:"7px"}}>
                  <Button className="PilotButton" id={driver[0]} onClick={() => this.props.handler(driver[0])}  fullWidth variant={driver[1].buttonVariant} color="primary">
                    {driver[0]}
                  </Button>
              	</Grid>
            ) 
          }
          </Grid> 
		</FormGroup>
	)}
}

export default PilotSelection;