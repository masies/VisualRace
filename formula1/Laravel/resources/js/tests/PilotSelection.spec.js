import React from 'react';
import { shallow, mount } from 'enzyme';
import PilotSelection from '../components/PilotSelection';

let drivers = {
  "Driver1": {
    name: "Driver1",
    data: {
      speed: [],
      rpm: [],
      throttle: [],
      coordinates: [],
      acceleration: [],
      coolant: [],
    },
    checked: true,
    buttonVariant: 'contained'
  }
}

let drivers2 = {
  "Driver2": {
    name: "Driver2",
    data: {
      speed: [],
      rpm: [],
      throttle: [],
      coordinates: [],
      acceleration: [],
      coolant: [],
    },
    checked: true,
    buttonVariant: 'outlined'
  }
}

const toggleButton = jest.fn()

let wrapped = mount(<PilotSelection handler={toggleButton} drivers={drivers} />);
let wrapped_shallow = shallow(<PilotSelection handler={toggleButton} drivers={drivers} />);

describe('PilotSelection', () => {
  it('should render the PilotSelection Component correctly', () => {
    expect(wrapped_shallow).toMatchSnapshot();
  });

  it('button click should toggle pilot selection', () => {

    wrapped.find('button').simulate('click')
    expect(toggleButton).toHaveBeenCalled()
  })

  it('button text should match pilots name', () => {
    expect(wrapped.find('button').text()).toEqual('Driver1')
  })

  it('should change buttons state when props change', () => {
    wrapped.setProps({ drivers: drivers2 })
    expect(wrapped.find('button').text()).toEqual('Driver2')

  })

});