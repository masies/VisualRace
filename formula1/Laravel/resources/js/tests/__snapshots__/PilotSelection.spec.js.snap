// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PilotSelection should render the PilotSelection Component correctly 1`] = `ShallowWrapper {}`;
